colcon_python_setup_py/package_identification/python_setup_py.py

```diff
@@ -244,7 +244,7 @@ def get_setup_information(setup_py, *, env):
     Dry run the setup.py file and get the configuration information.
 
     :param Path setup_py: path to a setup.py script
-    :param dict env: environment variables to set before running setup.py
+    :param Mapping[str, str] env: environment variables to set before running setup.py
```

Comment:
I don't think `Mapping` is a defined type in this context at the moment. Should it be `collections.abc.Mapping`?

Are there any cases where this is called with a `dict`?

Reply:
`typing.Mapping[str, str]`. If you try to use collections.abc, you'll get an error.
I don't know if there are cases now, but you want to allow a dict, so that it can be modified before passing. Changing os.environ is probably not a good idea.